src/components/messageListener.ts

```diff
@@ -2,8 +2,53 @@ import { Message } from 'discord.js';
 import { applyBonusByUserId } from './coin';
 import { client } from '../bot';
 import { TextChannel } from 'discord.js';
+import { logError } from './logger';
+import { vars } from '../../config/config';
+
+const PC_CATEGORY_ID: string = vars.PC_CATEGORY_ID;
+const HONEYPOT_CHANNEL_ID: string = vars.HONEYPOT_CHANNEL_ID;
+
+/*
+ * Detect spammers/trolls/people who got hacked, not by the honeypot method, by
+ * detecting that the message contains a ping and punishable word, and is sent
+ * in the Discord server, not in the PC category
+ */
+const detectSpammersAndTrollsNotByHoneypot = (message: Message): boolean => {
+  // Pings that would mention many people in the Discord server
+  const pingWords = ['@everyone', '@here'];
+  // Keywords that point towards the user being a spammer/troll/someone who got hacked
+  const punishableWords = ['http', 'nitro'];
+  return (
+    pingWords.some((word) => message.content.includes(word)) &&
+    punishableWords.some((word) => message.content.toLowerCase().includes(word)) &&
+    message.channel instanceof TextChannel &&
+    message.channel.parentID !== PC_CATEGORY_ID
+  );
+};
+
+/*
+ * Punish spammers/trolls/people who got hacked
+ */
+const punishSpammersAndTrolls = async (message: Message): Promise<boolean> => {
+  if (detectSpammersAndTrollsNotByHoneypot(message) || message.channel.id === HONEYPOT_CHANNEL_ID) {
+    // Temp ban member for 1 day and 1 hour, and delete messages from past 1 day
+    try {
+      if (await message.member?.ban({ days: 1, reason: 'Spammer/troll/got hacked' })) {
+        setTimeout(async () => await message.guild?.members.unban(message.author.id), 25 * 60 * 60 * 1000); // 1 day and 1 hour in milliseconds
```

Comment:
This implementation may not be resilient. Let's say if a user gets banned, and then the bot crashes or a new version gets deployed within 24 hours, then the user will not get unbanned.

Reply:
An alternative solution that I can think of is that the time of unban would be stored in the database and there would be a cron job to check if it is time to do any unbans? Also, is the current implementation okay or is a more resilient implementation wanted?

Reply:
A db isn't needed as long as we print out exactly who was banned in # moderation and when they were banned, prior to actually kicking them.

We can manually do unbans upon request, which is probably very rare anyways.
What if we KICKED instead of banning for some amt of time? Right now, I think msgs disappear, which might not always be good. Suppose I stopped being a mod, and became a norm member, and got hacked, and made spam, hence banned - then would my 4000+ msgs all disappear?